octodns_easydns/__init__.py

```diff
@@ -230,24 +260,72 @@ def _data_for_CNAME(self, _type, records):
     def _data_for_DS(self, _type, records):
         values = []
         for record in records:
-            key_tag, algorithm, digest_type, digest = record['rdata'].split(
-                ' ', 3
-            )
+            try:
+                key_tag, algorithm, digest_type, digest = record['rdata'].split(
+                    ' ', 3
+                )
+            except ValueError:
+                continue
             values.append(
                 {
-                    'key_tag': key_tag,
-                    'algorithm': algorithm,
-                    'digest_type': digest_type,
+                    'key_tag': int(key_tag),
+                    'algorithm': int(algorithm),
+                    'digest_type': int(digest_type),
                     'digest': digest,
                 }
             )
         return {'type': _type, 'values': values, 'ttl': int(records[0]['ttl'])}
 
+    def _data_for_SSHFP(self, _type, records):
+        values = []
+        for record in records:
+            try:
+                algorithm, fingerprint_type, fingerprint = record[
+                    'rdata'
+                ].split(' ', 2)
+            except ValueError:
+                continue
+
+            values.append(
+                {
+                    'algorithm': int(algorithm),
+                    'fingerprint': fingerprint,
+                    'fingerprint_type': int(fingerprint_type),
+                }
+            )
+        return {'ttl': int(records[0]['ttl']), 'type': _type, 'values': values}
+
+    def _data_for_TLSA(self, _type, records):
+        values = []
+        for record in records:
+            try:
+                (
+                    certificate_usage,
+                    selector,
+                    matching_type,
+                    certificate_association_data,
+                ) = record['rdata'].split(' ', 3)
+                values.append(
+                    {
+                        'certificate_usage': int(certificate_usage),
+                        'selector': int(selector),
+                        'matching_type': int(matching_type),
+                        'certificate_association_data': certificate_association_data,
+                    }
+                )
+            except ValueError:
+                continue
```

Comment:
And here, though in this one the appends seems to be inside the try/except so it'll never happen if there's a `ValueError`